### PR TITLE
Uniform jsonrpc type return

### PR DESCRIPTION
**Pull request by ferranbt**
# Description

This PR use the return types in the `jsonrpc` package instead of an explicit conversion inside the endpoint.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [ ] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### Manual tests

Please complete this section if you ran manual tests for this functionality, otherwise delete it

# Documentation update

Please link the documentation update PR in this section if it's present, otherwise delete it

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

File: jsonrpc/eth_endpoint.go [hunk 1]
```diff
@@ -433,12 +427,12 @@ func (e *Eth) GetStorageAt(
 
 // GasPrice returns the average gas price based on the last x blocks
 // taking into consideration operator defined price limit
-func (e *Eth) GasPrice() (string, error) {
+func (e *Eth) GasPrice() (interface{}, error) {
```
Comment (JekaMas) [inline jsonrpc/eth_endpoint.go:430]: why empty interface is better? what user cases it solves?

Comment (ferranbt) [reply jsonrpc/eth_endpoint.go:430]: That is how all the endpoints are defined. The `jsonrpc` library uses reflection to decide which methods export. The convention for all the methods is to return `(interface{}, error)`. This change just respects that convention.

**Review by Stefan-Ethernal — APPROVED** [review]
Overall looks good to me, although unit tests are failing, because it is needed to change assertions (due to changes in return types `string` vs `arg...` 
https://github.com/0xPolygon/polygon-edge/actions/runs/3401432822/jobs/5677755750

**Comment by ferranbt** [comment]
> Overall looks good to me, although unit tests are failing, because it is needed to change assertions (due to changes in return types `string` vs `arg...`

Thank you! I did fail before due to the multi-lock error so I missed this error.

**Review by vcastellm — APPROVED** [review]
LGTM

**Review by JekaMas — APPROVED** [review]

**Pull request merged**